src/setViewport.js

```diff
@@ -14,32 +15,40 @@ const MIN_VIEWPORT_SCALE = 0.0001;
 export default function (element, viewport) {
   const enabledElement = getEnabledElement(element);
 
-  enabledElement.viewport.translation.x = viewport.translation.x;
-  enabledElement.viewport.translation.y = viewport.translation.y;
-  enabledElement.viewport.voi.windowCenter = viewport.voi.windowCenter;
-  enabledElement.viewport.invert = viewport.invert;
-  enabledElement.viewport.pixelReplication = viewport.pixelReplication;
-  enabledElement.viewport.rotation = viewport.rotation;
-  enabledElement.viewport.hflip = viewport.hflip;
-  enabledElement.viewport.vflip = viewport.vflip;
-  enabledElement.viewport.modalityLUT = viewport.modalityLUT;
-  enabledElement.viewport.voiLUT = viewport.voiLUT;
-  enabledElement.viewport.colormap = viewport.colormap;
-  enabledElement.viewport.labelmap = viewport.labelmap;
+  // If viewport is not already set, start with default and merge new
+  // viewport options later
+  if (enabledElement.viewport === undefined) {
+    enabledElement.viewport = getDefaultViewport(enabledElement.canvas);
+  }
+
+  // Merge viewport
+  if (viewport) {
```

Comment:
Could we do this with Object.assign instead? I'm not sure what it does with null / undefined values.

Reply:
Now that I said that, I see that displayImage does the same thing (https://github.com/cornerstonejs/cornerstone/blob/master/src/displayImage.js#L41) so you probably copied it from there. No need to change it I guess.